table/server/common/src/main/java/alluxio/table/common/udb/UdbFilterSpec.java

```diff
@@ -0,0 +1,505 @@
+   * When this set is empty, mIgnoreMode is set to NONE.
+   */
+  private final Set<NameTemplate> mIgnoredTables;
+  /**
+   * Mode of ignored tables: included or excluded.
+   */
+  private final Mode mIgnoreMode;
+
+  private UdbFilterSpec(
+      Set<NameTemplate> bypassedTables,
+      Mode tableBypassMode,
+      Map<String, Pair<Mode, Set<NameTemplate>>> bypassedPartitions,
+      Set<NameTemplate> ignoredTables,
+      Mode ignoreMode) {
+    mBypassedTables = ImmutableSet.copyOf(bypassedTables);
+    mIgnoredTables = ImmutableSet.copyOf(ignoredTables);
+    mBypassedPartitions = ImmutableMap.copyOf(bypassedPartitions);
+
+    if (!mBypassedTables.isEmpty()) {
+      Preconditions.checkArgument(tableBypassMode != null && tableBypassMode != Mode.NONE,
+          "Missing inclusion/exclusion mode for bypassed tables");
+      mTableBypassMode = tableBypassMode;
+    } else {
+      // override mode to NONE when no tables are given
+      mTableBypassMode = Mode.NONE;
+    }
+    if (!mIgnoredTables.isEmpty()) {
+      Preconditions.checkArgument(ignoreMode != null && ignoreMode != Mode.NONE,
+          "Missing inclusion/exclusion mode for ignored tables");
+      mIgnoreMode = ignoreMode;
+    } else {
+      mIgnoreMode = Mode.NONE;
+    }
+    if (!mBypassedPartitions.isEmpty()) {
+      // forbid having partition spec when tables are excluded
+      Preconditions.checkArgument(mTableBypassMode != Mode.EXCLUDE,
+          "Tables are specified with exclusion mode, cannot have partition specification");
+      for (Map.Entry<String, Pair<Mode, Set<NameTemplate>>> entry :
+          mBypassedPartitions.entrySet()) {
+        String tableName = entry.getKey();
+        Mode partitionMode = entry.getValue().getFirst();
+        Set<NameTemplate> partitionNameTemplates = entry.getValue().getSecond();
+        // forbid empty table name
+        Preconditions.checkArgument(!tableName.isEmpty(),
+            "Empty table name specified with partition specification %s",
+            partitionNameTemplates);
+        // forbid empty spec
+        Preconditions.checkArgument(!partitionNameTemplates.isEmpty(),
+            "Empty partition specification set for table %s",
+            tableName);
+        // forbid NONE mode
+        Preconditions.checkArgument(partitionMode != Mode.NONE,
+            "Invalid mode %s for partitions of table %s, must be either include or exclude",
+            partitionMode, tableName);
+        // forbid both exact name and partition spec for a table at the same time
+        Preconditions.checkArgument(mBypassedTables.stream()
+                .noneMatch(template -> template instanceof ExactNameTemplate
+                    && template.matches(tableName)),
+            "Table %s is already specified with exact name, "
+                + "cannot have partition specification at the same time",
+            tableName);
+      }
+    }
+  }
+
+  public static UdbFilterSpec empty() {
```

Comment:
`/usr/src/alluxio/table/server/common/src/main/java/alluxio/table/common/udb/UdbFilterSpec.java:139:3: error: Missing a Javadoc comment.
`

Reply:
Added doc in cef77153696b4028a525829d6d17d898fa972e6c